src/Foundation/Commerce/code/Services/Order/OrderService.cs

```diff
@@ -15,31 +15,156 @@
         }
 
-        public Result<OrderHistoryResultModel> GetOrders(DateTime? fromDate, DateTime? untilDate, int page, int count)
+        public Result<IList<Order>> GetOrders(DateTime? fromDate, DateTime? untilDate, int page, int count)
         {
-            throw new NotImplementedException();
+            var result = new Result<IList<Order>>();
+
+            var headersResult = this.orderManager.GetOrdersHeaders(
+                this.visitorContext.ContactId,
+                this.storefrontContext.ShopName);
+
+            if (headersResult.Success)
+            {
+                IList<Order> orders = headersResult.OrderHeaders
+                    .Where(header => fromDate == null ? true : header.OrderDate >= fromDate)
+                    .Where(header => untilDate == null ? true : header.OrderDate <= untilDate)
+                    .Skip(page * count)
+                    .Take(count)
+                    .Select(
+                        header =>
+                        {
+                            var orderResult = this.orderManager.GetOrder(
+                                header.OrderID,
+                                this.visitorContext.ContactId,
+                                this.storefrontContext.ShopName);
+
+                            if (!orderResult.Success)
+                            {
+                                result.SetErrors(orderResult.SystemMessages.Select(sm => sm.Message).ToList());
+                            }
+
+                            return this.mapper.Map(orderResult.Order);
+                        })
+                    .ToList();
+
+                result.SetResult(orders);
+            }
+            else
+            {
+                result.SetErrors(headersResult.SystemMessages.Select(sm => sm.Message).ToList());
+            }
+
+            return result;
         }
 
-        public Result<SubmitOrderModel> SubmitOrder()
+        public Result<OrderConfirmation> SubmitOrder()
         {
-            throw new NotImplementedException();
+            var result = new Result<OrderConfirmation>();
+
+            var cartResult = this.cartManager.LoadCart(this.storefrontContext.ShopName, this.visitorContext.ContactId);
+
+            if (cartResult.Success)
+            {
+                var submirResult = this.orderManager.SubmitVisitorOrder(cartResult.Cart);
+
+                if (submirResult.Success)
+                {
+                    result.SetResult(
+                        new OrderConfirmation
+                        {
+                            ConfirmationId = submirResult.Order.TrackingNumber
+                        });
+                }
+                else
+                {
+                    result.SetResult(
+                        new OrderConfirmation
+                        {
+                            ConfirmationId = submirResult.Order.TrackingNumber,
+                            Temp = submirResult.CartWithErrors
```

Comment:
In previous implementation Temp property was used only in case of success. Do we need cart information on UI in case of error?

Reply:
This property is `TsIgnore` and can be accessed only as for error temp object.

Reply:
So far I haven't found usage of Temp on UI. Do we actually need it in our model?

Reply:
No, we don't  use this on UI, may be we could just log this `CartWithErrors` and delete this field